api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java

```diff
@@ -1,16 +1,120 @@
+                      "ApiException: Welcome Email to '%s' for user '%s' not sent: %s", contactEmail, user.getName(), attempt.getRight().toString()));
+                    if (retries == 0) {
+                        log.log(Level.SEVERE, String.format(
+                          "ApiException: On Last Attempt! Welcome Email to '%s' for user '%s' not sent: %s", contactEmail, user.getName(), attempt.getRight().toString()));
+                        throw new MessagingException("Sending email failed: " + attempt.getRight().toString());
+                    }
+                    break;
+
+                case REJECTED:
+                    log.log(Level.SEVERE, String.format(
+                      "Messaging Exception: Welcome Email to '%s' for user '%s' not sent: %s", contactEmail, user.getName(), attempt.getRight().toString()));
+                    throw new MessagingException("Sending email failed: " + attempt.getRight().toString());
+
+                case SUCCESSFUL:
+                    log.log(Level.INFO, String.format(
+                      "Welcome Email to '%s' for user '%s' was sent.", contactEmail, user.getName()));
+                    break retry;
+
+                default:
+                  log.log(Level.SEVERE, String.format(
+                    "Something went wrong when trying to send Welcome Email. This case signifies something wrong in the code."));
+                  if (retries == 0) {
+                  log.log(Level.SEVERE, String.format(
+                    "Welcome Email to '%s' for user '%s' was not sent.", contactEmail, user.getName()));
+                  }
+            }
+        } while (retries > 0);
+    }
+
+    private MandrillMessage buildWelcomeMessage(String contactEmail, String password, User user) {
+        MandrillMessage msg = new MandrillMessage();
+        RecipientAddress toAddress = new RecipientAddress();
+        toAddress.setEmail(contactEmail);
+        msg.setTo(Collections.singletonList(toAddress));
+        String msgBody = "Your new account is: " + user.getPrimaryEmail() +
+          "\nThe password for your new account is: " + password;
+        msg.setHtml(msgBody);
+        msg.setSubject("Your new All of Us Account");
+        msg.setFromEmail(workbenchConfigProvider.get().mandrill.fromEmail);
+        return msg;
+    }
+
+    private ImmutablePair<String, String> trySend(MandrillApiKeyAndMessage keyAndMessage) {
+        try {
+            log.log(Level.INFO, "trying send");
+            MandrillMessageStatuses msgStatuses = mandrillApiProvider.get().send(keyAndMessage);
+            for (MandrillMessageStatus msgStatus : msgStatuses) {
+                if (msgStatus.getRejectReason() != null) {
+                    return new ImmutablePair<String, String>("REJECTED", msgStatus.getRejectReason());
+                }
+            }
+        } catch (Exception e) {
+            log.log(Level.INFO, "in catch");
+            return new ImmutablePair<>("API_ERROR", e.toString());
```

Comment:
Do we need to distinguish between different kinds of errors / exceptions here?

Reply:
I would use the real `Status` here instead of doing string manipulation in other places.